src/gradient_slider.cpp

```diff
@@ -64,7 +66,20 @@ GradientSlider::GradientSlider(Qt::Orientation orientation, QWidget *parent) :
 
 GradientSlider::~GradientSlider()
 {
-	delete p;
+    delete p;
+}
+
+void GradientSlider::mousePressEvent(QMouseEvent *ev)
+{
+    bool vertical = orientation() == Qt::Vertical;
+    bool reverse = vertical != invertedAppearance();
```

Comment:
Looks like invertedAppearance appearance isn't fully supported (eg: the gradient slider doesn't invert the gradient) but at least it works so I guess it's OK.